bundles/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/handler/PLCBridgeHandler.java

```diff
@@ -256,20 +295,26 @@ public void dispose() {
         logger.debug("Dispose LOGO! bridge handler.");
         super.dispose();
 
+        final var rtcJob = this.rtcJob;
         if (rtcJob != null) {
             rtcJob.cancel(false);
-            rtcJob = null;
-            logger.info("Destroy RTC job for {}.", config.get().getAddress());
+            logger.info("Destroy RTC job for {}.", config.getAddress());
         }
+        this.rtcJob = null;
 
+        final var readerJob = this.readerJob;
         if (readerJob != null) {
             readerJob.cancel(false);
-            readerJob = null;
-            logger.info("Destroy reader job for {}.", config.get().getAddress());
+            logger.info("Destroy reader job for {}.", config.getAddress());
         }
+        this.readerJob = null;
 
-        if (disconnect()) {
-            client = null;
+        final var client = this.client;
+        if ((client != null) && client.isConnected()) {
```

Comment:
Seen this at many places. I stop now to comment on it, but i woudl suggest to not refactor these conditions like this.
```suggestion
        if (client != null && client.isConnected()) {
```